.claude/scripts/mcp_common.sh

```
@@ -0,0 +1,2094 @@
+        elif [ -n "$grok_default_model" ]; then
+            grok_env_flags+=(--env "XAI_DEFAULT_CHAT_MODEL=$grok_default_model")
+        fi
+
+        if [ -n "${XAI_MODEL:-}" ]; then
+            grok_env_flags+=(--env "XAI_MODEL=$XAI_MODEL")
+        elif [ -n "${XAI_DEFAULT_CHAT_MODEL:-}" ]; then
+            grok_env_flags+=(--env "XAI_MODEL=$XAI_DEFAULT_CHAT_MODEL")
+        elif [ -n "$grok_default_model" ]; then
+            grok_env_flags+=(--env "XAI_MODEL=$grok_default_model")
+        fi
+        add_cmd=(${MCP_CLI_BIN} mcp add "${MCP_SCOPE_ARGS[@]}" "${cli_args[@]}" "${grok_env_flags[@]}" "$name" "$NODE_PATH" "$grok_path" "${cmd_args[@]}")
```

Shellcheck (0.11.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)